combinetf2/fitter.py

```diff
@@ -179,6 +183,67 @@ def __init__(self, indata, options):
             and ((not self.binByBinStat) or self.binByBinStatType == "normal")
         )
 
+    def set_blinding(self, blind=True, unblind_parameter_expressions=[]):
+        self.blind = blind
+
+        def compile_patterns(patterns):
+            compiled = []
+            for p in patterns:
+                if p.startswith("r:"):
+                    # Treat as regex, remove prefix
+                    compiled.append(re.compile(p[2:]))
+                else:
+                    # Treat as exact string match
+                    compiled.append(re.compile(rf"^{re.escape(p)}$"))
+            return compiled
+
+        # Find parameters that match any regex
+        compiled_regexes = compile_patterns(unblind_parameter_expressions)
+        unblind_parameters = [
+            s
+            for s in [*self.indata.procs, *self.indata.noigroups]
+            if any(regex.search(s.decode()) for regex in compiled_regexes)
+        ]
+
+        self.unblind_parameters = unblind_parameters
+
+    def get_blinding_offsets(
+        self,
+    ):
+        # random blinding with seed taken based on string of parameter name
+
+        # backup and restore the initial random generator state
+        state = np.random.get_state()
+
+        def deterministic_random_from_string(s):
+            # Hash the string
+            hash = hashlib.sha256(s).hexdigest()
+            # Convert first 8 hex digits (32 bits) to int
+            seed = int(hash[:8], 16)
```

Comment:
This should use np.random.SeedSequence, which can probably directly consume int(hash, 16).  This less elegant version was used because the older version of numpy used in the combinetf1 environment did not have this functionality.